be/src/exprs/agg/distinct.h

```diff
@@ -176,13 +176,11 @@ struct DistinctAggregateStateV2<PT, FixedLengthPTGuard<PT>> {
 
         size_t old_size = set.size();
         src += sizeof(size);
-        const T* data = reinterpret_cast<const T*>(src);
-        static const size_t prefetch_dist = 4;
         for (size_t i = 0; i < size; i++) {
-            if ((i + prefetch_dist) < size) {
-                set.prefetch(data[i + prefetch_dist]);
-            }
-            set.insert(data[i]);
+            T key;
```

Comment:
Is it possiable use
```suggestion
            if consexpr (sizeof(T) == 16) {
               // memory copy
            } else {
              // reinterpret_cast
            }
```

Reply:
Yes, we can do that, but it's a little tricky. And if we do some undefined behavior , maybe there are other bugs besides unaligned load that will bite you in an unpredictable way. 

According to our past experience, the `memcpy` right here is just a hint to prevent the optimization bug and will not cause any perf degradation.

Reply:
case 1: https://dorisdb.atlassian.net/browse/SR-2702

case 2: https://github.com/StarRocks/starrocks/blob/main/be/src/formats/orc/apache-orc/c++/src/LzoDecompressor.cc#L65

Reply:
OK